app/Pasteboard.m

```diff
@@ -0,0 +1,258 @@
+#include <string.h>
+#import <UIKit/UIKit.h>
+#include "fs/poll.h"
+#include "fs/dyndev.h"
+#include "kernel/errno.h"
+#include "debug.h"
+
+// Prepare for fd separation
```

Comment:
Message to future self: this is explained in discord.

Reply:
re-posting it here:

What do you think about refactoring fd system a bit so fd data for different types of fd isn't placed in one union in one file, but instead is:
- replaced by fsprivate pointer (additional allocation for each fd, additional dereference for many operations), OR
- is placed after shared fd (I like this more, but it would require more refactoring i think)

For the second, additional field (fd_priv_size or smth) can be added to fd_opts, and fd_create would do malloc(sizeof(struct fd) + ops.fd_priv_size) instead of malloc(sizeof(struct fd)).

It'd be a great addition for dyn_dev, since right now I'd have to add clipboard related data into union in fd.h, but it really shouldn't be there.